massa-module-cache/src/controller.rs

```diff
@@ -55,16 +58,21 @@ impl ModuleCache {
                 warn!("compilation of module {} failed with: {}", hash, e);
                 ModuleInfo::Invalid
             }
-        }
+        };
+        let end = Instant::now();
+        let elapsed = end.duration_since(start).as_micros();
+        debug!("TIMER: cl compilation took {} μs", elapsed);
+        module
     }
 
     /// Save a new or an already existing module in the cache
     pub fn save_module(&mut self, bytecode: &[u8]) {
         let hash = Hash::compute_from(bytecode);
-        if let Some(hd_module_info) =
-            self.hd_cache
-                .get(hash, self.cfg.compilation_gas, self.cfg.gas_costs.clone())
-        {
+        if let Some(hd_module_info) = self.hd_cache.get(
+            hash,
+            self.cfg.gas_costs.max_instance_cost,
+            self.cfg.gas_costs.clone(),
```

Comment:
can we pass by reference to avoid cloning the whole thing ?

Reply:
I agree that this would be a nice change, but hd_cache requires the object because module deserializer requires it, which is later required by the engine init etc.

I'll open an issue on this & can take care of it later but it's a lot of changes in runtime & node for that to happen.